ArjunaJTA/jdbc/classes/com/arjuna/ats/internal/jdbc/ConnectionImple.java

```diff
@@ -379,7 +379,8 @@ public void close() throws SQLException
 
 	        if (!delayClose)  // close now
 	        {
-	            _transactionalDriverXAConnectionConnection.closeCloseCurrentConnection();
+	            if (_theModifier != null)
```

Comment:
Well, I have merged this now so it should close the regression I caused. What I propose is that we create a new bug to look into this case as it was not addressed before either. Would you like to raise the Jira? We should reference this Jira for hints if we don't tackle it straight away